### PR TITLE
Update to vulkan-headers, glslang. Submission vulkan-tools, vulkan-validationlayers

### PR DESCRIPTION
**Pull request by eintw1ck**
#### Description

These ports are independent of the MoltenVk issue as they're on a higher level where MoltenVk is just the driver (even though MoltenVk is a weird driver since you can link to it directly).

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.5 18F132
Xcode 10.2.1 10E1001

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ x squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by eintw1ck** [comment]
vulkan-headers, vulkan-tools, and vulkan-validationlayers all fail because they expect `python` to be `python3`, what's the best way to fix this? depend on python37, create a dir with symlink `python` -> `python37` and add it to path?

File: graphics/vulkan-headers/Portfile [hunk 1]
```diff
@@ -4,9 +4,9 @@ PortSystem          1.0
 PortGroup           github 1.0
 PortGroup           cmake 1.0
 
-github.setup        KhronosGroup Vulkan-Headers sdk-1.1.101.0
+github.setup        KhronosGroup Vulkan-Headers v1.1.112
 name                vulkan-headers
-version             1.1.101.0
+version             v1.1.112
```
Comment (mf2k) [inline graphics/vulkan-headers/Portfile]: We don't want the version to have the "v" in it. Remove the version line and change the `github.setup` line to be:

`github.setup        KhronosGroup Vulkan-Headers 1.1.112 v
`

File: graphics/vulkan-tools/Portfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        KhronosGroup Vulkan-Tools v1.1.112
+name                vulkan-tools
+version             v1.1.112
```
Comment (mf2k) [inline graphics/vulkan-tools/Portfile]: Same comment as above. Remove the `version` line entirely and use this instead:

`github.setup        KhronosGroup Vulkan-Tools 1.1.112 v
`

File: graphics/vulkan-tools/Portfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+homepage            https://vulkan.lunarg.com
+use_zip             no
+
+checksums           sha256  80e182fc21f4d8636e929243609956f2e9fb86ab5a1afd0cb9e0f10af59e1ac6 \
+                    rmd160  3375d9ef0bbb47d17b5be1d72a46078043313b5c \
+                    size    616030
+
+patchfiles          cmake.patch
+
+PortGroup           cmake 1.0
```
Comment (mf2k) [inline graphics/vulkan-tools/Portfile]: Use the cmake 1.1 portgroup instead.

File: graphics/vulkan-validationlayers/Portfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        KhronosGroup Vulkan-ValidationLayers v1.1.112
+name                vulkan-validationlayers
+version             v1.1.112
```
Comment (mf2k) [inline graphics/vulkan-validationlayers/Portfile]: As above, remove the `version` line and instead use:

`github.setup        KhronosGroup Vulkan-ValidationLayers 1.1.112 v
`

File: graphics/vulkan-validationlayers/Portfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+homepage            https://vulkan.lunarg.com
+use_zip             no
+
+checksums           sha256  8e9347ec3109438762bd62aabb5f463bda1b3a94bfb711649e8efc233c6eef17 \
+                    rmd160  58e4db9d0a7a8adedba4f740d93b0a1c2be36d7a \
+                    size    1194094
+
+patchfiles          cmake.patch
+
+PortGroup           cmake 1.0
```
Comment (mf2k) [inline graphics/vulkan-validationlayers/Portfile]: Use and test the cmake 1.1 portgroup here.

File: graphics/vulkan-validationlayers/Portfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        KhronosGroup Vulkan-ValidationLayers v1.1.112
+name                vulkan-validationlayers
+version             v1.1.112
+
+categories          graphics
+platforms           darwin
+license             Apache-License-2.0
```
Comment (mf2k) [inline graphics/vulkan-validationlayers/Portfile]: This is how we specify the Apache 2.0 license:

`license             Apache-2
`

**Comment by mf2k** [comment]
> vulkan-headers, vulkan-tools, and vulkan-validationlayers all fail because they expect `python` to be `python3`, what's the best way to fix this? depend on python37, create a dir with symlink `python` -> `python37` and add it to path?

I don't see any dependencies on python3. You probably need to at least add that.

**Comment by pmetzger** [comment]
@eintw1ck Please implement @mf2k's changes (and rebase/force push) so we can move your update forward. Thanks!

**Comment by eintw1ck** [comment]
@mf2k the vulkan-loader port requires moltenvk which isn't necessarily true (which makes it fail on xcode 7)--you could use an externally built moltenvk library instead (which is what I have been using). Is it worth removing the dependency? Also if there was another custom ICD other than moltenvk that could be used as well

**Comment by pmetzger** [comment]
@mf2k How does the changed version look to you?

**Comment by mf2k** [comment]
I briefly looked and I think it's good. Another review is preferred though. 

@pmetzger: Please merge if you approve too.

File: graphics/spirv-tools/Portfile [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+PortGroup           cmake 1.1
+
+github.setup        KhronosGroup SPIRV-Tools 2019.3 v
+name                spirv-tools
+
+categories          graphics
+platforms           darwin
+license             Apache-2
+
+maintainers         {itanimul.li:josh @eintw1ck}
```
Comment (mf2k) [inline graphics/spirv-tools/Portfile]: Can `openmaintainer` be added here?

File: graphics/vulkan-tools/Portfile [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+PortGroup           cmake 1.1
+
+github.setup        KhronosGroup Vulkan-Tools 1.1.112 v
+name                vulkan-tools
+
+categories          graphics
+platforms           darwin
+license             Apache-2
+
+maintainers         {itanimul.li:josh @eintw1ck}
```
Comment (mf2k) [inline graphics/vulkan-tools/Portfile]: Can `openmaintainer` be added here?

File: graphics/vulkan-tools/Portfile [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+description         Vulkan tools
+long_description    Provides vulkaninfo command for checking Vulkan installation configuration status
+
+depends_build-append port:vulkan-headers port:python37
```
Comment (mf2k) [inline graphics/vulkan-tools/Portfile]: [minor] This violates the mode line. Please adjust the spacing. Commonly in this case we add a slash at the end of the line and move the ports to the next line. Also, please only list one port per line and also sort them by name. So change this to the following:

```
depends_build-append        \
                            port:python37 \
                            port:vulkan-headers
```

File: graphics/vulkan-validationlayers/Portfile [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+PortGroup           cmake 1.1
+
+github.setup        KhronosGroup Vulkan-ValidationLayers 1.1.112 v
+name                vulkan-validationlayers
+
+categories          graphics
+platforms           darwin
+license             Apache-2
+
+maintainers         {itanimul.li:josh @eintw1ck}
```
Comment (mf2k) [inline graphics/vulkan-validationlayers/Portfile]: Can `openmaintainer` be added here?

File: graphics/vulkan-validationlayers/Portfile [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+description         Vulkan layers for validation and debugging
+long_description    Provides Vulkan validation layers that can be enabled to assist development by enabling developers to verify their applications correct use of the Vulkan API
+
+depends_build-append port:vulkan-headers port:python37
```
Comment (mf2k) [inline graphics/vulkan-validationlayers/Portfile]: Same comment as above with regards to the formatting of this line.

File: graphics/vulkan-validationlayers/Portfile [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+depends_lib-append  port:vulkan-loader port:glslang port:spirv-tools
```
Comment (mf2k) [inline graphics/vulkan-validationlayers/Portfile]: Please add one dependency per line and sort by name.

**Comment by pmetzger** [comment]
@eintw1ck If you could implement @mf2k's requests we should be able to merge this.

**Comment by eintw1ck** [comment]
@pmetzger implemented all changes

**Comment by mf2k** [comment]
Looking better. You just need to clean up the commit message so they all start with the port name. Like this:

For example, change:
`Add vulkan-tools port
`

To:
`vulkan-tools: new port`

**Comment by eintw1ck** [comment]
@mf2k commit messages were updated.

**Comment by pmetzger** [comment]
Cool. I think we can commit as soon as the CI completes.

**Comment by eintw1ck** [comment]
@pmetzger does it need to pass Xcode 7.3? I checked the log but couldn't see why it failed

**Pull request merged**